xcode/Subconscious/Shared/Models/Petname.swift

```diff
@@ -61,6 +61,14 @@ public struct Petname:
         "@\(self.verbatim)"
     }
     
+    public var leaf: Petname {
+        self.parts().first ?? self
+    }
+    
+    public var root: Petname {
+        self.parts().last ?? self
+    }
```

Comment:
Unsure on names but I sense this is an important concept

Reply:
Yes.

Per some of the bugs @jsantell found, I think in future, we should create a `Petname.Part` type that performs the same sanitization as Petname, but disallows `.`.

Petname should store,

```
let parts: [Part]
let leaf: Part
let root: Part
let description: String {
    parts.map({ part in part.description }).join(".")
}
```

...instead of storing a string.

Non-blocking. We can address in a follow-up.

Reply:
Yeah this feels like the right shape of solution to me, packing into and unpacking the string representation is wasteful and confusing.